### PR TITLE
feat: initial commit with atag

### PR DESCRIPTION
**Pull request by SoloJiang**
*Before* submitting a pull request, please make sure the following is done...

1. Fork the repo and create your branch from `master`.
2. If you've added code that should be tested, add tests!
3. If you've changed APIs, update the documentation.
4. Ensure the test suite passes (`npm test`).
5. Make sure your code lints (`npm run lint`) - we've done our best to make sure these rules match our internal linting guidelines.

### CODE REVIEW

File: packages/atag/vendors/native-shim.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  // including ES6 class definition
+  // iOS 8 can not recognize by throwing StynaxError
+  try {
+    eval('class A {}');
+  } catch (err) {
+    // not support es6 class
+    return;
+  }
+
+  try {
```
Comment (wssgcg1213) [inline packages/atag/vendors/native-shim.js]: need source code

File: packages/atag/vendors/polyfills.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+/**
+ * POLYFILL ES THAT ATAG USED.
+ * Array.prototype.includes
+ * Objcet.setPrototypeOf
+ * Object.assign
+ */
+
+if (!Array.prototype.includes) {
```
Comment (wssgcg1213) [inline packages/atag/vendors/polyfills.js]: 引用 runtime-shared

File: babel.config.js [hunk 3]
```diff
@@ -12,10 +12,11 @@ module.exports = function(api) {
     ],
     'plugins': [
       '@babel/plugin-proposal-export-default-from',
+      ['@babel/plugin-proposal-decorators', {'legacy': true}],
       ['@babel/plugin-proposal-class-properties', { 'loose': false }],
       'babel-plugin-export-default-module-exports',
       'babel-plugin-transform-jsx-stylesheet',
-      'babel-plugin-add-module-exports'
+      'babel-plugin-add-module-exports',
```
Comment (boiawang) [inline babel.config.js]: 没必要逗号

File: packages/atag/README.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+npm run build # build to dist/atag.js
+```
+
+测试
+
+```bash
+npm test
+```
+
+* [组件文档](https://developer.taobao.com/components)
```
Comment (boiawang) [inline packages/atag/README.md]: 地址打不开？

File: packages/atag/config/webpack.config.base.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+const { resolve } = require('path');
```
Comment (boiawang) [inline packages/atag/config/webpack.config.base.js]: 这些 webpack 统一提到外面的 scripts 下面会不会更好？

File: packages/atag/config/webpack.config.dev.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const getBase = require('./webpack.config.base');
+const merge = require('webpack-merge');
+// const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
```
Comment (boiawang) [inline packages/atag/config/webpack.config.dev.js]: 既然没用就去掉吧

File: packages/atag/src/shared/utils/debounce.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
+var isObject = require('./isObject'),
```
Comment (boiawang) [inline packages/atag/src/shared/utils/debounce.js]: utils 下面基本都是 lodash 里面的功能，为啥不直接 import 进来？

File: packages/atag/README.md (hunk 4 above)
Comment (SoloJiang) [reply packages/atag/README.md]: 恩，这里要改下，最后要加个/

**Pull request closed without merging**